web/models/survey.ex

```diff
@@ -427,6 +434,141 @@ defmodule Ask.Survey do
     }
   end
 
+  def retries_histograms(%Survey{mode: mode} = survey) do
```

Comment:
It may pay off to delegate all the histograms calculations to a different module, and to heavily test its component functions.

Instead of testing the whole histogram, it's easier to test (and understand) what each individual function is calculating, and how they cooperate to get to the final result.